src/components/content/content-blocks/link-selector/link-selector.tsx

```diff
@@ -64,18 +64,24 @@ export const LinkSelector = ({
 		);
 	}
 	return (
-		<Popover
-			type={linkType}
-			id={node.target}
-			forcePopover={nodeWithQuery}
-			query={getQueryUrl(node)}
-			onExit={() => onExit()}
-			side={textDirection === "ltr" ? "right" : "left"}
-			trigger={getTriggerComp(linkType, componentData, className)}
-			portalled={true}
+		<span
+			data-test-annotation-type={node.linkType}
+			data-test-target={node.target.split(`/`)[1]}
+			data-test-seq={node.sequence}
```

Comment:
Ahem? Do you intend to leave test-specific markup in the component, or is this temp?

Reply:
I agree it's a price to pay; the alternative was integrating Stylable into the test runtime in order to capture the actual classname. Using data attributes is a common practice; although again, I agree - not optimal. Will look for another alternative.

Reply:
(all three comments refer to this issue)
OK, I don't know what alternatives exist, so can't contribute anything. Generally,  I won't approve any intrusive solution, no matter how minor and graceful the intrusion is. If you end up deciding that this is the best way to go, just remove me from the approval cycle.